docs/application/web/guides/data/file-system.md

```diff
@@ -108,149 +107,261 @@ To manage file storages:
+           /* Display the file path with name */
+           console.log('File path and name is: ' + files[i]);
+       }
+   }
+   function onerror(error) {
+       console.log(error);
+   }
+   tizen.filesystem.listDirectory('documents/subDir', onsuccess, onerror);
    ```
 
-4. To retrieve the file content as a DOMString, use the `readAsText()` method.
-
-   The encoding input parameter of the method defines the format in which the file content is returned.
+2. To retrieve the file URI, use the `toURI()` method:
 
    ```
-           if (files[i].isDirectory == false) {
-               files[i].readAsText(function(str) {
-                   console.log('File content: ' + str);
-               }, null, 'UTF-8');
-           }
+   function onsuccess(files) {
+       for (var i = 0; i < files.length; i++) {
+           /* Display the file URI */
+           console.log('File URI is: ' + toURI(files[i]));
        }
    }
+   function onerror(error) {
+       console.log(error);
+   }
+   tizen.filesystem.listDirectory('documents/subDir', onsuccess, onerror);
    ```
 
+3. To retrieve the file content as a DOMString, binary data or to put it's contents to a `Blob` object use one of the following functions on `FileHandle` object opened in modes appropriate for reading ('r', 'rw').
+
+   - First open a file using `openFile()` function with mode allowing to read it's contents.
+      ```
+      /* Opening file for read - this code assumes that there is */
+      /* a file named 'file' in documents directory */
+      var fileHandleRead = tizen.filesystem.openFile('documents/file', 'r');
+      console.log('File opened for reading');
+      ```
+
+   - Use readString() or readStringNonBlocking() to retrieve the file content as a DOMString.
+      ```
+      var fileContents = fileHandleRead.readString();
+      console.log('File contents: ' + fileContents);
+      ```
+
+   - Use readData() or readDataNonBlocking() to retrieve the file content as a Uint8Array object (binary data).
+      ```
+      var fileContents = fileHandleRead.readData();
+      console.log('File binary contents in array:');
+      console.log(fileContents);
+      ```
+
+   - Use readBlob() or readBlobNonBlocking() to retrieve the file content as a Blob object.
+      ```
+      var fileContents = fileHandleRead.readBlob();
+      console.log('Blob object:');
+      console.log(fileContents);
+      /* FileReader is a W3C API class, not related to webapi-plugins */
+      /* and is capable of extracting blob contents */
+      var reader = new FileReader();
+      /* Event fires after the blob has been read/loaded */
+      reader.addEventListener('loadend', function(contents)
+      {
+         const text = contents.srcElement.result;
+         console.log('File contents: ' + text);
+      });
+      /* Start reading the blob as text */
+      reader.readAsText(fileContentsInBlob);
+      ```
+
+   - When file is no longer needed, remember to close it.
```

Comment:
Please rephrase to:

Close the file when it is not in use.

Reply:
Done.